src/test/java/fi/otavanopisto/kuntaapi/test/ResourceMocker.java

```diff
@@ -0,0 +1,194 @@
+      .willReturn(aResponse()
+      .withStatus(404));
+    
+    resources.put(id, new MockedResource<>(resource, okMapping, notFoundMapping));
+  }
+  
+  public Collection<MockedResource<R>> getMockedResources() {
+    return resources.values();
+  }
+  
+  public Collection<MockedResource<R>> getMockedResources(MockedResourceStatus status) {
+    Collection<MockedResource<R>> result = new ArrayList<>();
+    
+    for (MockedResource<R> resource : getMockedResources()) {
+      if (resource.getStatus().equals(status)) {
+        result.add(resource);
+      }
+    }
+    
+    return result;
+  }
+  
+  public List<R> getResources(MockedResourceStatus status) {
+    List<R> result = new ArrayList<>();
+    
+    for (MockedResource<R> mockedResource : getMockedResources(status)) {
+      result.add(mockedResource.getResource());
+    }
+    
+    return result;
+  }
+  
+  public boolean isMocked(I id) {
+    return resources.containsKey(id);
+  }  
+  
+  public void setStatus(I id, MockedResourceStatus status) {
+    MockedResource<R> resource = resources.get(id);
+    if (resource.getStatus() == status) {
+      return;
+    }
+    
+    if (started) {
+      if (resource.getCurrentMapping() != null) {
+        removeStub(resource.getCurrentMapping());
+      }
+    }
+        
+    resource.setStatus(status);
+    
+    if (started) {
+      stubFor(resource.getCurrentMapping());
+      updateStatusLists();
+      
+      if (subMockers.containsKey(id)) {
+        for (ResourceMocker<?, ?> subMocker : subMockers.get(id)) {
+          if (status == MockedResourceStatus.OK) {
+            subMocker.start();
+          } else {
+            subMocker.stop();
+          }
+        }
+      }
+    }
+  }
+
+  public void addStatusList(MockedResourceStatus status, UrlPattern urlPattern) {
+    statusLists.put(status, get(urlPattern));
+  }
+
+  public void addSubMocker(I id, ResourceMocker<?, ?> subMocker) {
+    if (!subMockers.containsKey(id)) {
+      subMockers.put(id, new ArrayList<>());
+    }
+    
+    subMockers.get(id).add(subMocker);
+  }
+
+  public ResourceMocker<?, ?> getSubMocker(I id, int index) {
+    return subMockers.get(id).get(index);
+  }
+  
+  private void updateStatusLists() {
+    for (Entry<MockedResourceStatus, MappingBuilder> statusListEntry : statusLists.entrySet()) {
+      MappingBuilder mapping = statusListEntry.getValue();  
+      MockedResourceStatus status = statusListEntry.getKey();
+      
+      removeStub(mapping);
+      
+      mapping.willReturn(aResponse()
+        .withHeader(CONTENT_TYPE, APPLICATION_JSON)
+        .withBody(toJSON(getResources(status))));
+      
+      stubFor(mapping);
+    }
+  }
+  
+  private String toJSON(Object object) {
+    try {
+      ObjectMapper objectMapper = new ObjectMapper();
+      objectMapper.registerModule(new JavaTimeModule());
+      return objectMapper.writeValueAsString(object);
+    } catch (JsonProcessingException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)